enterprise/management/commands/ensure_singular_active_enterprise_customer_user.py

```diff
@@ -0,0 +1,107 @@
+"""
+Django management command to ensure there is at most a single EnterpriseCustomerUser
+with `active=True` for each enterprise user.
+"""
+import logging
+from time import sleep
+
+from django.apps import apps
+from django.core.management import BaseCommand
+from django.db.models import Count
+
+from enterprise.utils import batch
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Django management command to ensure there is at most a single EnterpriseCustomerUser
+    with `active=True` for each enterprise user.
+
+    Example usage:
+    ./manage.py lms ensure_singular_active_enterprise_customer_user
+    """
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--batch-limit',
+            action='store',
+            dest='batch_limit',
+            default=100,
+            help='Number of records in each batch.',
+            type=int,
+        )
+
+        parser.add_argument(
+            '--batch-sleep',
+            action='store',
+            dest='batch_sleep',
+            default=5,
+            help='How long to sleep between batches.',
+            type=int,
+        )
+
+    def _enterprise_customer_user_model(self):
+        """ Returns `EnterpriseCustomerUser model """
+        EnterpriseCustomerUser = apps.get_model(
+            'enterprise',
+            'EnterpriseCustomerUser',
+        )
+        return EnterpriseCustomerUser
+
+    def _historical_enterprise_customer_user_model(self):
+        """ Returns `HistoricalEnterpriseCustomerUser model """
+        HistoricalEnterpriseCustomerUser = apps.get_model(
+            'enterprise',
+            'HistoricalEnterpriseCustomerUser',
+        )
+        return HistoricalEnterpriseCustomerUser
+
+    def _process_user(self, user):
+        """
+        Finds all EnterpriseCustomerUser objects for the specified user, flipping all but one to
+        have `active=False`. The one that is not flipped (remains `active=True`) is based on the
+        historical records of which EnterpriseCustomerUser object was flipped to `active=True`
+        most recently.
+        """
+        EnterpriseCustomerUser = self._enterprise_customer_user_model()
+        HistoricalEnterpriseCustomerUser = self._historical_enterprise_customer_user_model()
+        lms_user_id = user['user_id']
+        current_active_ecu_count = user['ecu_count']
+        log.info('Processing LMS User ID %s', lms_user_id)
+        active_historical_ecus = HistoricalEnterpriseCustomerUser.objects.filter(user_id=lms_user_id, active=True)
+        most_recent_active_ecu = active_historical_ecus.order_by('-history_date').first()
```

Comment:
fyi I think you can use the `.history` manager on the enterprise customer user to do something similar
```
my_enterprise_customer_user_object.history.filter(active=True)
```
really just a "btw", in case you wanted to approach it differently 🤷

Reply:
actually nevermind, there are multiple enterprise customer user objects, scratch this

Reply:
@christopappas So, the last time I worked with historical models (I think also in a management command), I had first resorted to the approach you're suggesting above since that's what's in the official documentation for django-simple-history. However, I had run into some odd error then (blanking on exactly what it was) so I just stuck with the current approach that worked before here.

Reply:
okay cool, np